# Тестовое задание. Шагдуров Баир

Подкинули монету N раз. Кол-во случаев, когда выпал орёл, на 10% больше, чем кол-во случаев, когда выпала решка. При каком N мы можем сказать, что монета «нечестная» (орёл и решка выпадают с разной вероятностью)?

## Ответ

В данном случае мы можем использовать Хи-квадрат Пирсона (даже несмотря на то, что величины Objected и Expected у орла
и решки - зависимые величины).

Дано: N, N(орел) = N * 0.6, N(решка) = N * 0.4. Найти количество N, когда мы можем отклонить нулевую гипотезу H0 (H0 - P(орла) = 0.5, альтернативная гипотеза - H1 - P(орла) != 0.5)

Для наглядности вычислений нарисуем таблицу:

![image.png](image.png)

In [2]:
#функция для нахождения значения Хи-квадрат Пирсона для заданного N (две степени свободы)
#(количество выпадений орла и решки задано заранее)
def HiSquare(N):
    ObjO = N * 0.6
    ObjR = N * 0.4
    Exp = N/2
    HiSq = (ObjO - Exp)**2/Exp + (ObjR - Exp)**2/Exp 
    return HiSq

Теперь нам необходимо узнать, при каком значении Хи-квадрат Пирсона мы достигнем необходимого нам уровня значимости. Возьмем стандартный уровень 0.05.

![image.png](image1.png)

Как видно из таблицы, для числа степеней свободы k = 2 необходимое значение Хи-квадрат Пирсона равняется 6.0

Это значит, что вероятность получить значения Хи-квадрат Пирсона равное или большее, чем 6.0 - < 5%

In [3]:
HiSqDict = dict()

Начнем перебор значений с N=100 с шагом 5

In [4]:
for N in range(100,200,5):
    HiSqDict[N] = HiSquare(N)

In [5]:
HiSqDict

{100: 4.0,
 105: 4.2,
 110: 4.4,
 115: 4.6,
 120: 4.8,
 125: 5.0,
 130: 5.2,
 135: 5.4,
 140: 5.6,
 145: 5.8,
 150: 6.0,
 155: 6.2,
 160: 6.4,
 165: 6.6,
 170: 6.8,
 175: 7.0,
 180: 7.2,
 185: 7.4,
 190: 7.6,
 195: 7.8}

Получен следующий результат:
При N >= 150 и выполнении условий задачи (количество выпадений орла на 10% больше, чем количество выпадений решки)
мы можем сказать, что монета "нечестная".

### Ответ: N >= 150 